plugin/clang.vim

```diff
@@ -904,6 +914,8 @@ func! s:ClangExecute(root, clang_options, line, col)
   " longer version, redirect output to different files
   let l:command = l:cmd.' 1>'.l:tmps[0].' 2>'.l:tmps[1]
   let l:res = [[], []]
+  let b:clang_state['stdout'] = l:res[0]
+  let b:clang_state['stderr'] = l:res[1]
```

Comment:
I don't know why move them to here. They should be assigned when in sync mode, and useless in async mode in this function. So I think both of them should be kept where they were.